packages/sign_in_with_apple/android/src/main/kotlin/com/aboutyou/dart_packages/sign_in_with_apple/SignInWithApplePlugin.kt

```diff
@@ -145,32 +168,61 @@ public class SignInWithApplePlugin: FlutterPlugin, MethodCallHandler, ActivityAw
  *
  * DO NOT rename this or it's package name as it's configured in the consumer's `AndroidManifest.xml`
  */
-public class SignInWithAppleCallback: Activity {
-  constructor() : super()
+public class SignInWithAppleCallback : Activity() {
+  private fun validateNonce( uri: Uri, expectedNonce: String?): Boolean {
+    val idToken = uri.getQueryParameter("id_token")
+    if (idToken == null) {
+      Log.e(TAG, "Missing id_token query parameter in signinwithapple callback")
+      return false
+    }
+
+    try {
+      val jwt = JWT(idToken)
+
+      if (jwt.getClaim("nonce_supported").asBoolean() == true) {
+        if (jwt.getClaim("nonce").asString() != expectedNonce) {
```

Comment:
I guess this warrants a comment, that we don't want to early return in case of the `expectedNonce` being `null` as we still check that the incoming data also doesn't have one.

Reply:
I wouldn't add any default nonce as this might lead to unexpected login failures in case someone tries to validate it (Firebase does, for example).

I would instead provide a small function for generating this nonce so the users don't have to write it themselves.

Reply:
```suggestion
        // Check that the given nonce matched the expected one, even if none is expected
        if (jwt.getClaim("nonce").asString() != expectedNonce) {
```

Reply:
Hmm, with the additional check below (https://github.com/aboutyou/dart_packages/pull/113/files#r464235837), I don't think we would even go into this branch if the expected nonce was `null`. So that comment doesn't make sense.